### ISSUE TITLE
username without a ssh-key or password

### ISSUE THREAD

**Issue by jiata** [Azure/thunderbolt#104]
My cluster.yaml is left as default (which sets `username: spark`)

When i run `azb spark cluster create --id ignitedemo`, it shows:
```
-------------------------------------------
spark cluster id:        ignitedemo
spark cluster size:      2
>        dedicated:      2
>     low priority:      0
spark cluster vm size:   standard_a2
path to custom script:   None
docker repo name:        jiata/thunderbolt:0.1.0-spark2.2.0-python3.5.4
wait for cluster:        True
username:                spark
-------------------------------------------
Waiting for all nodes in pool ignitedemo to reach desired state...
```
So this is confusing for the user that doesn't want to use an ssh key (windows users). It's supposedly creating a username `spark`, but with what password? 

**I think it should prompt for a password if no ssh is set, or we should not, by default, prefill the username field in the config**

Then i get this error, which is good - but it is only thrown after the cluster is created:
```
dtde.error.InvalidUserCredentialsError: Cannot add user to cluster. Need to provide a ssh public key or password.
```

**Comment by jafreck** [comment]
This should be handled by validating the input. Probably an error should be thrown if you try to create a cluster with a username but no ssh-key or password. Prompting for a password would be even better, like you said.

**Comment by jafreck** [comment]
Fixed by #116
